api/endpoints/conversations.py

```diff
@@ -0,0 +1,96 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponse
+from api.models import Conversation, Message
+from api.common import *
+
+# Conversation-related endpoints reside here.
+
+# URI: /api/conversations
+# Expect:
+# Response: array of {Role (PiN or Volunteer), UID, Name, ProfilePic, TruncatedLastMessage}
+def conversations(request):
+    if "uemail" not in request.session:
+        return httpBadRequest()
+    response = []
+
+    convos_pin = None
```

Comment:
In a better world, we'd have a method of determining the type of user (pin or volunteer) given their id, but oh well :)

Reply:
Hmm, I don't know if it's that simple. Remember a volunteer can also convert to being a PiN using the slider button. So we have to simultaneously allow for a user to be a PiN or Volunteer in a conversation. They can both coexist on a single account.

Reply:
Oh, true. I forgot that was a thing. In that case this function will return this user's all conversations. (as a pin and volunteer). I guess it makes more sense to show all messages. 

Thanks.